spec/triggers/README.md

```diff
@@ -1,11 +1,107 @@
-# KEDA Triggers
+# Trigger Specification
 
-This section describes all the specifications for all supported triggers in KEDA.
+This specification describes the `trigger` section of the `ScaledObject` used to define how what triggers KEDA should use to scale your application.
+
+<details>
+  <summary><b>Table of Contents</b></summary>
+
+- [Metadata](#metadata)
+- [Authentication](#authentication)
+    - [Environment variable(s)](#environment-variables)
+    - [Secret(s)](#secrets)
+    - [Azure Pod Identity](#azure-pod-identity)
+- [Supported Trigger Typess](#supported-trigger-types)
+</details>
+
+[`types.go`](./../pkg/apis/keda/v1alpha1/types.go)
+
+```yaml
+type: {trigger-type} # Required.
+metadata:
+    # {list of properties to configure a trigger}
+authentication:
```

Comment:
While documenting this I was considering if we have to scope `env` & `secretTargetRef` to `authentication` or if we should introduce another section like `parameters` or something similar.

Reply:
However, keep in mind that metadata already has trigger specific configuration, so might confuse/conflict with that